packages/scripts/src/lib/_migrate-glintrc.ts

```diff
@@ -0,0 +1,302 @@
+
+const EnvironmentMap = z.record(
+  z.object({
+    additionalGlobals: z.optional(z.array(z.string())),
+  })
+);
+type EnvironmentMap = z.infer<typeof EnvironmentMap>;
+
+const Environment = z.optional(z.union([z.string(), EnvironmentList, EnvironmentMap]));
+const Files = z.optional(z.array(z.string()));
+
+const GlintRc = z.object({
+  environment: Environment,
+  checkStandaloneTemplates: z.optional(z.boolean()),
+  include: Files,
+  exclude: Files,
+});
+
+type GlintRc = z.infer<typeof GlintRc>;
+
+type GlintConfig = {
+  environment?:
+    | string
+    | string[]
+    | Record<
+        string,
+        {
+          additionalGlobals?: string[] | undefined;
+        }
+      >;
+  checkStandaloneTemplates?: boolean;
+  transform?: {
+    include?: string[];
+    exclude?: string[];
+  };
+};
+
+function loadFile(path: string): Promise<Result<string, string>> {
+  return readFile(path, { encoding: 'utf-8' })
+    .then((v) => ok<string, string>(v))
+    .catch((e) => err(`Could not load file at ${path}: ${JSON.stringify(e)}`));
+}
+
+function loadOrCreateTsconfig(configPath: string): Promise<string> {
+  return readFile(configPath, { encoding: 'utf-8' }).catch((e) => {
+    console.info(
+      `Could not load tsconfig.json at ${configPath}: ${e}; attempting to create a new one`
+    );
+    return '{}';
+  });
+}
+
+function saveFile(path: string, data: string): Promise<Result<Unit, string>> {
+  return writeFile(path, data)
+    .then(() => ok<Unit, string>())
+    .catch((e) => err(`Could not write file at ${path}: ${JSON.stringify(e)}`));
+}
+
+function assert(predicate: unknown, reason: string): asserts predicate {
+  if (!predicate) {
+    throw new Error(`panic: ${reason}`);
+  }
+}
+
+function parseGlintRcFile(contents: string): Result<GlintRc, string> {
+  let errors: string[] = [];
+  let yamlData = yaml.load(contents, {
+    onWarning: (yamlException) => errors.push(yamlException.toString()),
+  });
+  if (errors.length) {
+    return err(`Could not parse YAML:\n\t${errors.join('\n\t')}`);
+  }
+
+  return tryOrElse(
+    (e) => {
+      assert(e instanceof z.ZodError, 'error somehow not a zod error');
+      return `Could not parse data as a Glint config:\n\t${e.format()._errors.join('\n\t')}`;
+    },
+    () => {
+      return GlintRc.parse(yamlData);
+    }
+  );
+}
+
+// Represents the part of a tsconfig.json file we care about. That is: not much!
+type ReadyToTransform = {
+  glint?: GlintConfig;
+};
+
+function loadTsconfig(contents: unknown): Result<ReadyToTransform, string> {
+  if (typeof contents !== 'string') {
+    return err(`Could not patch ${JSON.stringify(contents)}: not a string`);
+  }
+
+  return tryOrElse(
+    (e) => `Could not patch data:\n\t${e instanceof Error ? e.message : JSON.stringify(e)}`,
+    () => {
+      let result = evaluate(contents);
+
+      if (typeof result !== 'object' || result == null || Array.isArray(result)) {
+        throw new Error('invalid contents of tsconfig.json file');
```

Comment:
Should maybe think about returning the actual contents so people can see what exactly it was that failed?